charactersheet/charactersheet/viewmodels/character/features.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+function FeaturesViewModel() {
+    var self = this;
+
+    self.DESCRIPTION_MAX_LENGTH = 45;
+
+    self.sorts = {
+        'name asc': { field: 'name', direction: 'asc'},
+        'name desc': { field: 'name', direction: 'desc'},
+        'description asc': { field: 'description', direction: 'asc'},
+        'description desc': { field: 'description', direction: 'desc'}
+    };
+
+    self.features = ko.observableArray([]);
+    self.blankFeature = ko.observable(new Feature());
+    self.blankTracked = ko.observable(new Tracked());
+    self.modalOpen = ko.observable(false);
+    self.editItemIndex = null;
+    self.currentEditItem = ko.observable(new Feature());
+    self.currentEditTracked = ko.observable(new Tracked());
+    self.sort = ko.observable(self.sorts['name asc']);
+    self.filter = ko.observable('');
+    self.shouldShowDisclaimer = ko.observable(false);
+    self.previewTabStatus = ko.observable('active');
+    self.editTabStatus = ko.observable('');
+    self.firstModalElementHasFocus = ko.observable(false);
+    self.editFirstModalElementHasFocus = ko.observable(false);
+
+    self.load = function() {
+        Notifications.global.save.add(self.save);
+
+        var key = CharacterManager.activeCharacter().key();
+        self.features(PersistenceService.findBy(Feature, 'characterId', key));
+    };
+
+    self.unload = function() {
+        Notifications.global.save.remove(self.save);
+    };
+
+    self.save = function() {
+        self.features().forEach(function(e, i, _) {
+            e.save();
+        });
+    };
+
+    // Pre-pop methods
+    self.featuresPrePopFilter = function(request, response) {
+        var term = request.term.toLowerCase();
+        var keys = DataRepository.features ? Object.keys(DataRepository.features) : [];
+        var results = keys.filter(function(name, idx, _) {
```

Comment:
We need to filter through every object and check the `displayName` field. There are no keys in this file. Or we change the data to have a key be the `displayName`.